### PR TITLE
docs: details about batching user operations

### PR DESCRIPTION
**Pull request by denniswon**
### Detailed summary

- Updated the documentation for batch transactions in Smart Contract Accounts.
- Added information about `ISmartContractAccount` implementation and `encodeBatchExecute` method.
- Added notes about batching transactions as a single User Operation and the ordering of batched transactions.
- Updated the example code for batching transactions using `sendUserOperation`.
- Updated the section heading for batching transactions using `sendTransactions`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by moldy530 — APPROVED** [review]

File: site/guides/batching-transactions.md [hunk 1]
```diff
@@ -20,16 +20,28 @@ head:
...
+One benefit of Smart Contract Accounts is that it's possible to batch transactions in one User Operation. Not all Smart Contract Accounts support batching. But, if the `SmartContractAccount` you're using does by implementing the [`encodeBatchExecute`](/packages/aa-core/accounts/optional/encodeBatchExecute.md) method, then implementations of `SmartAccountProvider` will allow you to make those calls.
 
 There are two ways you can batch transactions using `SmartAccountProvider`:
 
 1. via `sendUserOperation`
 2. via `sendTransactions`
 
-## `sendUserOperation`
+:::tip Note 1: `SimpleSmartContractAccount` and `LightSmartContractAccount`
+Both `SimpleSmartContractAccount` and `LightSmartContractAccount` implements `encodeBatchExecute`, thus supports batching `UserOperations` out of the box.
```
Comment (avasisht23) [inline site/guides/batching-transactions.md:31]:
```suggestion
Both `SimpleSmartContractAccount` and `LightSmartContractAccount` implement `encodeBatchExecute`, thus supports batching `UserOperations` out of the box.
```

File: site/guides/batching-transactions.md [hunk 2]
```diff
@@ -20,16 +20,28 @@ head:
 
 # How to Submit Batch Transactions
 
-One benefit of Smart Contract Accounts is that it's possible to batch transactions in one User Operation. Not all Smart Contract Accounts support batching. But, if the account you're using does, then implementations of `SmartAccountProvider` will allow you to make those calls.
+One benefit of Smart Contract Accounts is that it's possible to batch transactions in one User Operation. Not all Smart Contract Accounts support batching. But, if the `SmartContractAccount` you're using does by implementing the [`encodeBatchExecute`](/packages/aa-core/accounts/optional/encodeBatchExecute.md) method, then implementations of `SmartAccountProvider` will allow you to make those calls.
```
Comment (avasisht23) [inline site/guides/batching-transactions.md:23]:
```suggestion
One benefit of Smart Contract Accounts is that it's possible to batch transactions in one User Operation. Not all Smart Contract Accounts support batching. But, if the `ISmartContractAccount` implementation you're using has the [`encodeBatchExecute`](/packages/aa-core/accounts/optional/encodeBatchExecute.md) method, then implementations of `SmartAccountProvider` will allow you to make those calls.
```

File: site/guides/batching-transactions.md [hunk 3]
```diff
@@ -20,16 +20,28 @@ head:
...
+:::tip Note 1: `SimpleSmartContractAccount` and `LightSmartContractAccount`
+Both `SimpleSmartContractAccount` and `LightSmartContractAccount` implements `encodeBatchExecute`, thus supports batching `UserOperations` out of the box.
+:::
+
+:::tip Note 2: Transactions Batched as a Single User Operation
+When you batch transactions, the transaction actions (`target`s and `calldata`s) are batched into a single `UserOperation`, where the sender is the account itself.
+:::
+
+:::tip Note 3: Batched Transactions Ordering
+The batched `UserOperation` gets executed by the account calling `executeBatch` method on [`SimpleAccount`](https://github.com/eth-infinitism/account-abstraction/blob/ver0.6.0/contracts/samples/SimpleAccount.sol) or [`LightAccount`](https://github.com/alchemyplatform/light-account/blob/v1.0.2/src/LightAccount.sol) smart contracts. `executeBatch` processes the input array of transactions data linearly, guaranting the execution order of those transactions to be **sequential**.
```
Comment (avasisht23) [inline site/guides/batching-transactions.md:39]:
```suggestion
The batched `UserOperation` gets executed by the account calling the `executeBatch` method on the [`SimpleAccount`](https://github.com/eth-infinitism/account-abstraction/blob/ver0.6.0/contracts/samples/SimpleAccount.sol) or [`LightAccount`](https://github.com/alchemyplatform/light-account/blob/v1.0.2/src/LightAccount.sol) smart contracts. `executeBatch` processes the input array of transactions data linearly, guaranteeing the execution order of those transactions to be **sequential**.
```

**Review by avasisht23 — APPROVED** [review]
small nits, let's fix before merging

**Pull request merged**